src/background.js

```diff
@@ -0,0 +1,124 @@
+import { app, protocol, BrowserWindow, ipcMain,} from 'electron'; // eslint-disable-line
+import { createProtocol, installVueDevtools } from 'vue-cli-plugin-electron-builder/lib';
+import logger from 'electron-log';
+import { scrape, SCRAPERS } from './service/scrapers.service';
+import { encryptProperty, decryptProperty } from "./service/encryption/credentials.service";
+import './store'; 
+
+const isDevelopment = process.env.NODE_ENV !== 'production';
+
+// Keep a global reference of the window object, if you don't, the window will
+// be closed automatically when the JavaScript object is garbage collected.
+let mainWindow;
+
+logger.info(`Welcome to ${app.name} log`);
+logger.info(`Version: ${app.getVersion()}`);
+
+const onError = (error) => logger.error(error);
+logger.catchErrors({ onError });
+global.logger = logger;
+// Scheme must be registered before the app is ready
+protocol.registerSchemesAsPrivileged([
+  { scheme: 'app', privileges: { secure: true, standard: true } },
+]);
+
+function onProgress({ percent }, step) {
+  mainWindow.webContents.send('onProgress-message', percent, step)
+}
+
+ipcMain.on('getScrapers', (event) => {
+  event.reply('getScrapers-reply', SCRAPERS);
+});
+ipcMain.on('encryptProperty', async (event, importer, fields) => {
+  event.returnValue = await encryptProperty(importer, fields);
+});
+ipcMain.on('decryptProperty', (event, importer, fields) => {
+  decryptProperty(importer, fields).then((decrypted) => {
+    event.reply('decryptProperty-reply', decrypted)
+  })
+});
+
+ipcMain.on('scrape', (event, installPath, scraperName, loginFields, showBrowser, logger) => {
+  scrape(installPath, scraperName, loginFields, showBrowser, onProgress).then((result) => {
+    event.reply('scrape-reply', result)
+  })
+});
```

Comment:
All these methods should be in a separate file.
But I think we will do that in another PR.

Reply:
sure